extension/src/lib/chromeEvented.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2016 RtF-US
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+/* eslint-env node */
+/* global fluid, require */
+
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+// TODO: Get rid of this repeated code by using the ContextAwareness API
+var chrome = chrome || fluid.require("sinon-chrome", require, "chrome");
+
+fluid.defaults("gpii.chrome.eventedComponent", {
```

Comment:
Positioning this as a mixin is a bit odd, since it creates a constantly increasing cleanup challenge. We can carry on like this, as long as we also add "onDestroy" listeners to do cleanup, but a more realistic design might be to simply create one single singleton component representing Chrome and its events, register it using "resolveRoot" and then bind to its events from where they are required with {gpii.chrome} or whatever context we choose. This would mean that the Infusion component design more closely matched the underlying reality.

Reply:
It turns out a bunch of components bind to various chrome events. It seemed more practical for each component to bind to their own chrome events and clean up after themselves. I re-worked the grade to take in an `eventRelayMap` to setup and remove the listener bindings. I've gone through and updated the other components to make use of this. The only cases that aren't handled are ones that bind to events from an instance of something, like a port.